addons/project/models/project_project.py

```diff
@@ -857,29 +857,25 @@ def _get_hide_partner(self):
         return False
 
     @api.model
-    def _create_analytic_account_from_values(self, values):
-        company = self.env['res.company'].browse(values.get('company_id', False))
+    def _get_values_analytic_account_batch(self, project_vals):
         project_plan_id = int(self.env['ir.config_parameter'].sudo().get_param('analytic.analytic_plan_projects'))
-
         if not project_plan_id:
             project_plan, _other_plans = self.env['account.analytic.plan']._get_all_plans()
             project_plan_id = project_plan.id
+        companies = self.env['res.company'].browse([project.get('company_id', False) for project in project_vals])
```

Comment:
i'd suggest renaming the 'project' into 'val', or at least 'project_val'. Using 'project' implicitly imply its a project record imo, while its just a dict of vals.